### PR TITLE
Take usage of URI fragments

### LINKED ISSUE #170: link to open cinny to a certain room
#### Is your feature request related to a problem? Please describe.
element has links like `https://app.element.io/#/room/#cinny:matrix.org`

#### Describe the solution you'd like
something like `https://app.cinny.in/#/room/#cinny:matrix.org`

#### Describe alternatives you've considered
no

#### Additional context
i like to send people links to join rooms on matrix with Good Client(TM)
cinny having a nice link thing to sign up and join a room would be epic

### PR DESCRIPTION
**Pull request by C0ffeeCode**
### Description

It would be handy if we could launch Cinny opening a room automatically.
This would also enable [the matrix.to project](https://github.com/matrix-org/matrix.to) to add support for Cinny if they would like to do so.

To preserve privacy and room ids etc. to be leaked to the hosting service unnecessarily this leverages URI fragments over URI queries or paths.
Paths and queries are being sent to the server, which is not required in this scenario, while fragments are not.

Element handles this "issue" similarly.

### Schema
dev.cinny.in/
- #/room/[roomid]
- #/room/[roomid]/[eventid]
- #/[spaceid]
- #/[spaceid]/[roomid]
- #/join/[roomid]

Fixes #170
Depends on #401

#### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

Preview: https://627cc9bb10a88f43bce051f0--pr-cinny.netlify.app
⚠️ Exercise caution. Use test accounts. ⚠️

### CODE REVIEW

**Comment by 3nt3** [comment]
[I created a PR](https://github.com/matrix-org/matrix.to/pull/286) to add cinny to the list of clients in https://matrix.to but it would be better UX if you could directly go the room instead of copy-pasting it into the client.

BTW: how should we handly copyright attribution of the cinny logo?

**Comment by theotheroracle** [comment]
as this pr's dependency was merged, is there anything it's currently waiting on ?

File: src/util/uriFragments.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+listeners.push(
+  window.addEventListener('hashchange', handleUriFragmentChange),
+
+  navigation.on(cons.events.navigation.ROOM_SELECTED, (selectedRoom, _previousRoom, eventId) => {
+    const pieces = window.location.hash.split('/');
+    if (!eventId) pieces.length = 3;
+    else pieces[3] = eventId;
+    if (!selectRoom) pieces.length = 2;
```
Comment (greentore) [inline src/util/uriFragments.js]: Typo. Should be:
`if (!selectedRoom) pieces.length = 2;`

**Comment by fnwbr** [comment]
Thank you a lot for the continued efforts on this.

I just tested the netlify-hosted version of this PR, and it's shaping up really nicely!
Some things that I noticed that don't work:

- Using "back" and "forward" history buttons in the browser don't do anything
- Opening a new window / new session of Cinny and navigating to a room right away, e.g. by clicking on a link such as https://411--pr-cinny.netlify.app/#/room/#cinny:matrix.org will show you the default welcome screen instead of the room you intended to navigate to

**Comment by aceArt-GmbH** [comment]
> * Opening a new window / new session of Cinny and navigating to a room right away, e.g. by clicking on a link such as https://411--pr-cinny.netlify.app/#/room/#cinny:matrix.org will show you the default welcome screen instead of the room you intended to navigate to

https://github.com/cinnyapp/cinny/pull/411/files#diff-cc58d91e2c920c0129c91191387a190b861bb84e12775c6f729ddcda65ce1f1fR40-R44 seems to be the problem. No code after is executed

With it removed, "back" and "forward" work too

cc #70 #108

**Comment by kfiven** [comment]
Closing in favour of #1633 adding the URL navigation.

**Pull request closed without merging**